cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/schedule/constraint/ConcurrencyConstraintTest.java

```diff
@@ -75,18 +72,20 @@ public void testMaxConcurrentRuns() {
 
     // add a run for the schedule
     Map<String, String> systemArgs = ImmutableMap.of(ProgramOptionConstants.SCHEDULE_NAME, schedule.getName());
-    store.setStart(WORKFLOW_ID, pid1, System.currentTimeMillis(), null, EMPTY_MAP, systemArgs);
+    store.setStartAndRun(WORKFLOW_ID, pid1, System.currentTimeMillis(), System.currentTimeMillis() + 1,
+                         EMPTY_MAP, systemArgs);
     assertSatisfied(true, concurrencyConstraint.check(schedule, constraintContext));
 
     // add a run for the program from a different schedule. Since there are now 2 running instances of the
     // workflow (regardless of the schedule name), the constraint is not met
     systemArgs = ImmutableMap.of(ProgramOptionConstants.SCHEDULE_NAME, "not" + schedule.getName());
-    store.setStart(WORKFLOW_ID, pid2, System.currentTimeMillis(), null, EMPTY_MAP, systemArgs);
+    store.setStartAndRun(WORKFLOW_ID, pid2, System.currentTimeMillis(), System.currentTimeMillis() + 1,
+                         EMPTY_MAP, systemArgs);
     assertSatisfied(false, concurrencyConstraint.check(schedule, constraintContext));
 
     // add a run for the program that wasn't from a schedule
     // there are now three concurrent runs, so the constraint will not be met
-    store.setStart(WORKFLOW_ID, pid3, System.currentTimeMillis(), null, EMPTY_MAP, EMPTY_MAP);
+    store.setStartAndRun(WORKFLOW_ID, pid3, System.currentTimeMillis(), 1);
```

Comment:
`System.currentTimeMillis() + 1`? or you can just initialize local variables `startTs` and `runTs` at the beginning and let them be shared by all the `setStartAndRun` since the time here doesn't matter